crates/ruff_python_ast/src/comparable.rs

```diff
@@ -689,8 +689,7 @@ pub struct ExprIf<'a> {
 
 #[derive(Debug, PartialEq, Eq, Hash)]
 pub struct ExprDict<'a> {
-    keys: Vec<Option<ComparableExpr<'a>>>,
-    values: Vec<ComparableExpr<'a>>,
+    items: Vec<(Option<ComparableExpr<'a>>, ComparableExpr<'a>)>,
```

Comment:
nit: I think it could be useful to have `ComparableDictItem` but it's not strictly required, so I'm fine with this option as well

Reply:
Implemented in https://github.com/astral-sh/ruff/pull/11267/commits/65e5b1dd4fd67d6570cf6ed1a7a9c3fa899105d4